wp-includes/class-wp-service-workers.php

```diff
@@ -152,7 +152,7 @@ public function serve_request() {
 		$file_hash = md5( $output );
 		@header( "ETag: $file_hash" ); // phpcs:ignore Generic.PHP.NoSilencedErrors.Discouraged, WordPress.PHP.NoSilencedErrors.Discouraged
 
-		$etag_header = isset( $_SERVER['HTTP_IF_NONE_MATCH'] ) ? trim( $_SERVER['HTTP_IF_NONE_MATCH'] ) : false;
+		$etag_header = isset( $_SERVER['HTTP_IF_NONE_MATCH'] ) ? trim( sanitize_text_field( $_SERVER['HTTP_IF_NONE_MATCH'] ) ) : false;
```

Comment:
This is unnecessary as the value is being used entirely as part of a comparison. And core is not doing this:

```
src/wp-admin/load-styles.php:47:if ( isset( $_SERVER['HTTP_IF_NONE_MATCH'] ) && stripslashes( $_SERVER['HTTP_IF_NONE_MATCH'] ) === $wp_version ) {
src/wp-admin/load-scripts.php:48:if ( isset( $_SERVER['HTTP_IF_NONE_MATCH'] ) && stripslashes( $_SERVER['HTTP_IF_NONE_MATCH'] ) === $wp_version ) {
src/wp-includes/class-wp.php:461:			if ( isset( $_SERVER['HTTP_IF_NONE_MATCH'] ) ) {
src/wp-includes/class-wp.php:462:				$client_etag = wp_unslash( $_SERVER['HTTP_IF_NONE_MATCH'] );
src/wp-includes/ms-files.php:65:$client_etag = isset( $_SERVER['HTTP_IF_NONE_MATCH'] ) ? stripslashes( 
 $_SERVER['HTTP_IF_NONE_MATCH'] ) : false;
```

What is missing however is `wp_unslash()` and the variable name is wrong:

```suggestion
		$if_none_match = isset( $_SERVER['HTTP_IF_NONE_MATCH'] ) ? trim( wp_unslash( $_SERVER['HTTP_IF_NONE_MATCH'] ) ) : false;
```